contracts/TokenConverter/AbstractTokenConverter.sol

```diff
@@ -0,0 +1,787 @@
+        internal
+        view
+        returns (
+            uint256 amountConvertedMantissa,
+            uint256 amountOutMantissa,
+            uint256 tokenInToOutConversion
+        )
+    {
+        if (amountInMantissa == 0) {
+            revert InsufficientInputAmount();
+        }
+
+        ConversionConfig memory configuration = conversionConfigurations[tokenAddressIn][tokenAddressOut];
+
+        if (!configuration.enabled) {
+            revert ConversionConfigNotEnabled();
+        }
+
+        uint256 tokenInUnderlyingPrice = priceOracle.getPrice(tokenAddressIn);
+        uint256 tokenOutUnderlyingPrice = priceOracle.getPrice(tokenAddressOut);
+
+        /// amount of tokenAddressOut after including incentive
+        uint256 conversionWithIncentive = MANTISSA_ONE + configuration.incentive;
+        /// conversion rate after considering incentive(conversionWithIncentive)
+
+        amountOutMantissa =
+            (amountInMantissa * tokenInUnderlyingPrice * conversionWithIncentive) /
+            (tokenOutUnderlyingPrice * EXP_SCALE);
+        amountConvertedMantissa = amountInMantissa;
+
+        tokenInToOutConversion = (tokenInUnderlyingPrice * conversionWithIncentive) / tokenOutUnderlyingPrice;
+    }
+
+    /// @notice To get the amount of tokenAddressIn tokens sender would send on receiving amountOutMantissa tokens of tokenAddressOut
+    /// @dev This function retrieves values without altering token prices.
+    /// @param amountOutMantissa Amount of tokenAddressOut user wants to receive
+    /// @param tokenAddressIn Address of the token to convert
+    /// @param tokenAddressOut Address of the token to get after conversion
+    /// @return amountConvertedMantissa Amount of tokenAddressOut should be transferred after conversion
+    /// @return amountInMantissa Amount of the tokenAddressIn sender would send to contract before conversion
+    /// @custom:error InsufficientInputAmount error is thrown when given input amount is zero
+    /// @custom:error ConversionConfigNotEnabled is thrown when conversion is disabled or config does not exist for given pair
+    function _getAmountIn(
+        uint256 amountOutMantissa,
+        address tokenAddressIn,
+        address tokenAddressOut
+    )
+        internal
+        view
+        returns (
+            uint256 amountConvertedMantissa,
+            uint256 amountInMantissa,
+            uint256 tokenInToOutConversion
+        )
+    {
+        if (amountOutMantissa == 0) {
+            revert InsufficientOutputAmount();
+        }
+
+        ConversionConfig memory configuration = conversionConfigurations[tokenAddressIn][tokenAddressOut];
+
+        if (!configuration.enabled) {
+            revert ConversionConfigNotEnabled();
+        }
+
+        uint256 tokenInUnderlyingPrice = priceOracle.getPrice(tokenAddressIn);
+        uint256 tokenOutUnderlyingPrice = priceOracle.getPrice(tokenAddressOut);
+
+        /// amount of tokenAddressOut after including incentive
```

Comment:
But this is an exact output query, so incentive should be applied to amountInMantissa AFAIU

Reply:
` 
  amountOutMantissa =
            (amountInMantissa * tokenInUnderlyingPrice * conversionWithIncentive) /
            (tokenOutUnderlyingPrice * EXP_SCALE);
`
The parameter **conversionWithIncentive** is currently being multiplied across the entire expression. I believe the outcome would remain unaffected if we were to first multiply **amountInMantissa** by **conversionWithIncentive** before proceeding with the multiplication involving other factors such as **tokenInUnderlyingPrice** and **tokenOutUnderlyingPrice**.

Reply:
Yeah, that's correct, I was only concerned about the comment here, which was fixed in 8d8f8c786e598bf59b0d90edda59f8df16be84af